tests/unit_tests/sql_parse_tests.py

```diff
@@ -271,6 +271,7 @@ def test_extract_tables_illdefined() -> None:
     assert extract_tables("SELECT * FROM catalogname..tbname") == {
         Table(table="tbname", schema=None, catalog="catalogname")
     }
+    assert extract_tables('SELECT * FROM "tbname') == set()
```

Comment:
The ill-formed SQL statement throws a `TokenError` as opposed to a `ParseError` in `sqlglot`.

Reply:
@tobymao ideally should illformed SQL throw a `TokenError` or `ParseError`?

Reply:
it depends on if it's a token error or a parser error!

when you have unbalanced quotes, it's basically impossible to finish tokenization, so that's why we throw a token error.

most everything else is a parser error